packages/sitecore-jss-nextjs/src/revalidate/revalidate-middleware.ts

```diff
@@ -0,0 +1,304 @@
+   */
+  clientFactory: GraphQLRequestClientFactory;
+  /**
+   * Whether multisite is configured
+   * Default is false
+   */
+  multiSite?: boolean;
+  /**
+   * Whether personalization is configured
+   * Default is false
+   */
+  personalize?: boolean;
+};
+
+const SITE_PREFIX = '_site_';
+const PERSONALIZE_PREFIX = '_variantId_';
+
+/**
+ * Middleware / handler for on-demand ISR (e.g. '/api/revalidate').
+ */
+export class RevalidateMiddleware {
+  private personalizeService: GraphQLPersonalizeService;
+
+  constructor(protected config: RevalidateConfig) {
+    this.personalizeService = new GraphQLPersonalizeService({
+      clientFactory: config.clientFactory,
+    });
+  }
+
+  /**
+   * Gets the Next.js API route handler
+   * @returns route handler
+   */
+  public getHandler(): (req: NextApiRequest, res: NextApiResponse) => Promise<void> {
+    return this.handler;
+  }
+
+  protected isEmpty(data: UpdatedPaths[]) {
+    return data.length === 0;
+  }
+
+  /**
+   * Extracts the paths from the updated paths
+   * @param {UpdatedPaths[]} filteredUpdates Updated paths
+   * @returns {string[]} paths
+   */
+  protected extractPaths(filteredUpdates: UpdatedPaths[]) {
+    return filteredUpdates.map((update) => update.path);
+  }
+
+  /**
+   * Gets the site name from the path name
+   * @param {string} pathname Path name
+   * @returns {string} site name
+   */
+  protected getSiteName(pathname: string) {
+    let siteName = '';
+
+    const path = pathname.endsWith('/') ? pathname : pathname + '/';
+    const result = path.match('(.*?)\\/');
+
+    if (result && result[1] !== '') {
+      siteName = result[1];
+    }
+
+    return siteName;
+  }
+
+  /**
+   * Gets the path name from the full path
+   * @param {string} fullPath Full path
+   * @returns {string} path name
+   */
+  protected getPathName(fullPath: string) {
+    const pathParts = fullPath.split('/').filter((part) => part !== '');
+
+    if (pathParts.length >= 2) {
+      const siteName = `/${pathParts[0]}/`;
+      const path = `/${pathParts.slice(1).join('/')}`;
+      return path.startsWith(siteName) ? path.slice(siteName.length) : path;
+    }
+
+    return '/';
+  }
+
+  protected extractSiteName(path: string): string {
+    const siteName: string = path.split('/')[0];
+    return siteName;
+  }
+
+  /**
+   * Filters out the updated paths and language from the request body
+   * @param {NextApiRequest} req Next.js API request
+   * @returns {UpdatedPaths[]} updated paths
+   */
+  protected getFilteredUpdates(req: NextApiRequest): UpdatedPaths[] {
+    if (!req.body?.updates || this.isEmpty(req.body.updates)) {
+      return [];
+    }
+
+    return req.body?.updates
+      .filter(
+        (update: Entity) =>
+          update.entity_definition === EntityDefinition.LayoutData && update.entity_culture
+      )
+      .map((update: Entity) => {
+        return {
+          path: update.identifier,
+          language: update.entity_culture,
```

Comment:
This just made me think about i18n routing... Do we need to also add language to the revalidate path if this is configured (our default)? I couldn't find anything documented in Next.js docs, but I did find some discussions.
https://github.com/vercel/next.js/discussions/40489

Reply:
Hmm to my understanding the path would still be the same. I check the webhook response and I get the following response.
![image](https://github.com/Sitecore/jss/assets/89087450/a86e2940-a7bc-4399-aa02-f590d2813bc4)